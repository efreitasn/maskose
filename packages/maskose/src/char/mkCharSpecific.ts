import { MaskCharPredicateFn } from '.';

export const MASK_CHAR_SPECIFIC_TYPE = 'MASK_CHAR_SPECIFIC_TYPE';

export type MaskCharSpecificObj = {
  type: typeof MASK_CHAR_SPECIFIC_TYPE;
  char: string;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A specific character expected to be in the value to be masked
 * @param char Character that will be used in the mask
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharSpecific(char: string, predicateFn?: MaskCharPredicateFn): MaskCharSpecificObj {
  return {
    type: MASK_CHAR_SPECIFIC_TYPE,
    char,
    predicateFn
  };
}