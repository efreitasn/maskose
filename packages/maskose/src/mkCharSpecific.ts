import { MaskCharPredicateFn } from './mkCharNum';

const MASK_CHAR_SPECIFIC_TYPE = 'MASK_CHAR_SPECIFIC_TYPE';

export type MkCharSpecificObj = {
  type: typeof MASK_CHAR_SPECIFIC_TYPE;
  char: string;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A specific character expected to be in the value to be masked.
 * @param char Character that will be used in the mask
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharSpecific(char: string, predicateFn?: MaskCharPredicateFn): MkCharSpecificObj {
  return {
    type: MASK_CHAR_SPECIFIC_TYPE,
    char,
    predicateFn
  };
}