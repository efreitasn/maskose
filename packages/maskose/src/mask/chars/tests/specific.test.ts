import mkCharSpecific from '../specific';

test('should match the object returned by mkCharSpecific()', () => {
  const result = mkCharSpecific('E');

  expect(result).toMatchSnapshot();
});

test('should throw an error when an string with length > 1 is provided', () => {
  const str = '20';

  expect(() => mkCharSpecific(str)).toThrow();
});