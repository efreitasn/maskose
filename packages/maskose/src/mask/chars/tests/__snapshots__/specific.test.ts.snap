// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the object returned by mkCharSpecific() 1`] = `
Object {
  "char": "E",
  "regExp": /E/,
  "repetitions": 1,
  "type": "MASKOSE_CHAR_SPECIFIC_TYPE",
  "valueLengthConditions": Array [],
}
`;
